probe_src/probe_py/cli.py

```diff
@@ -116,4 +116,86 @@ def dump(
                     print(op.data)
                 print()
 
+@app.command(
+    context_settings=dict(
+        ignore_unknown_options=True,
+    ),
+)
+def ssh(
+        ssh_args: list[str],
+        debug: bool = typer.Option(default=False, help="Run verbose & debug build of libprobe"),
+):
+    """
+    Wrap SSH and record provenance of the remote command.
+    """
+
+    if "-i" not in ssh_args:
+        typer.secho("RSA key not defined", fg=typer.colors.RED)
+        raise typer.Abort()
+
+    rsa_index = ssh_args.index("-i")
+    rsa = ssh_args[rsa_index + 1]
+
+    # Determine the correct library to use
+
+    libprobe = project_root / "libprobe/build" / ("libprobe-dbg.so" if debug else "libprobe.so")
+    if not libprobe.exists():
+        typer.secho(f"Libprobe not found at {libprobe}", fg=typer.colors.RED)
+        raise typer.Abort()
+
+    # Create a temporary directory on the local machine
+    local_temp_dir = pathlib.Path(tempfile.mkdtemp(prefix=f"probe_log_{os.getpid()}"))
+
+    # Check if remote platform matches local platform
+    remote_gcc_machine_cmd = ["ssh"] + ssh_args + ["gcc", "-dumpmachine"]
+    local_gcc_machine_cmd = ["gcc", "-dumpmachine"]
+
+    remote_gcc_machine = subprocess.check_output(remote_gcc_machine_cmd).decode().strip()
+    local_gcc_machine = subprocess.check_output(local_gcc_machine_cmd).decode().strip()
+
+    if remote_gcc_machine != local_gcc_machine:
+        raise NotImplementedError("Remote platform is different from local platform")
+
+    if "-u" not in ssh_args:
+        typer.secho("Remote host not defined", fg=typer.colors.RED)
+        raise typer.Abort()
+
+    remote_index = ssh_args.index("-u")
+    remote_host = ssh_args[remote_index + 1]
+
+    ssh_cmd = ["ssh","-i",rsa,remote_host,"-p","2222"]
+
+    # Upload libprobe.so to the remote temporary directory
+    remote_temp_dir_cmd = ssh_cmd + ["mktemp", "-d", "/tmp/probe_log_XXXXXX"]
+    remote_temp_dir = subprocess.check_output(remote_temp_dir_cmd).decode().strip()
+    remote_probe_dir = f"{remote_temp_dir}/probe_dir"
+
+    scp_cmd = ["scp","-i",rsa, "-P", "2222", str(libprobe), f"{remote_host}:{remote_temp_dir}/"]
+    subprocess.run(scp_cmd, check=True)
```

Comment:
I think the best way to "pass the SSH arguments to SCP" is to run `ssh <ssh_args> -G`, capture the output (should be a list of pairs delimited by newlines and space), then send those to `scp -o <ssh_pair> -o <ssh_pair> ...`